sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -175,6 +175,30 @@ namespace Azure { namespace Core { namespace Http {
     NetworkAuthenticationRequired = 511, ///< HTTP 511 Network Authentication Required.
   };
 
+  /**
+   * @brief Defines a range of bytes within an HTTP resource, starting at an `Offset` and ending at
+   * `Offset + Length - 1` inclusively.
+   *
+   */
+  struct Range
+  {
+    /**
+     * @brief The starting point of the HTTP Range.
+     *
+     */
+    int64_t Offset = 0;
+
+    /**
+     * @brief The size of the HTTP Range.
+     *
+     */
+    int64_t Length = 0;
+  };
+
+  namespace Internal {
+    std::string RangeToString(Range range);
```

Comment:
If you have `ToString` here, you'll have to add `FromString`. Or you can just remove both of them.

I prefer you remove both

Reply:
Why do we have to add `FromString`? Is there a need?

Reply:
Well, I guess we don't

Reply:
> Why do we have to add FromString? Is there a need?

It is currently there on Storage code. That's why I added here.  The Range needs to become an HTTP header at the end. That's why it needs to become string.

@Jinming-Hu  will the Server ever return a Range header that we need to parse?

Reply:
Well, yes and not. Server returns something like `ContentRange: bytes=100-110/200". But storage team can parse in our own code.

Reply:
@vhvb1989, to clarify I asked about `FromString`, **not** `ToString`.

> It is currently there on Storage code.

Where?

> That's why I added here. 

Where? I only see `RangeToString`.

> The Range needs to become an HTTP header at the end. That's why it needs to become string.

Right, `ToString` is fine, keep it, I was posing the question to Jinming that just because we have one (internal) method, doesn't mean we need to add its counter. :)

> Server returns something like `ContentRange: bytes=100-110/200". But storage team can parse in our own code.

From what I can gather content range and this range type we are adding are not exactly the same thing and don't have the same format necessarily either (no equal sign, for example `Content-Range: bytes 200-1000/67589`).

Reply:
I will remove toString.
It might end up to be different for each service

Reply:
There is no reason to remove ToString. It is useful to have one standard (most commonly used) string representation of HttpRange that most callers will use. Any service that wants to change the behavior and create there own can always create their own helper and write it differently. By adding one for `HttpRange` were aren't preventing them from doing that.
Consider seeing if any service in .NET overrides the HttpRange.ToString value and how common that is. We can pick one format that is common enough to work for N services, and let the rest override.

Reply:
Given that the storage is the only one currently asking for this, I won't add the toString() for now.
Since this is not exposed to public API, we can decide later if we want to add it.

I think an end-user doesn't really need to ever call toString() on the Range.  It is only the Service the one who needs it for adding a header.
As an end-user, I could expect to get something like `range={offset: x, length: y}` for toString() because we are adapting `toString()` for the format as it is expected to be as part of an HTTP header.  We are also hardocding the `bytes` which is not cool.

So, makes more sense not to add it for now